Integrations/src/main/java/io/deephaven/integrations/learn/Scatterer.java

```diff
@@ -0,0 +1,57 @@
+package io.deephaven.integrations.learn;
+
+import io.deephaven.datastructures.util.CollectionUtil;
+import io.deephaven.integrations.python.PythonFunction;
+import org.jpy.PyObject;
+
+import java.util.ArrayList;
+
+/**
+ * Applies scatter functions to the result of Future.get() and feeds the scattered data back into a table.
+ */
+public class Scatterer {
+
+    private final Output[] outputs;
+
+    /**
+     * Creates a new Scatterer.
+     *
+     * @param outputs array of Outputs that determine how data will be scattered back to the table.
+     */
+    public Scatterer(Output ... outputs) { this.outputs = outputs; }
+
+    /**
+     * Applies the scatter function to a subset of result.
+     *
+     * @param result        result of the call to Future.get() containing data to be scattered.
+     * @param scatterFunc   function that determines how result will be parsed and fed into the table.
+     * @param offset        offset from FutureOffset that gets correct row from result.
+     * @return subset of result that can be put back into the table.
+     */
+    public PyObject scatter(PyObject result, PyObject scatterFunc, long offset) {
```

Comment:
1) This class isn't really doing the scatter.  This class is just getting the value to be scattered.  Think about the docs and the name.
2) Would the code be cleaner and faster if you had a signature of `PyObject scatter(Output output, FutureOffset fo)`? Does this change resolve the Future permissions?

Reply:
See comment below.  I'm guessing the target signature is really: `PyObject scatter(int idx, FutureOffset fo)`.